### PR TITLE
Add logic type tests

### PR DESCRIPTION
**Pull request by kmaehashi**
Implements #901.

### CODE REVIEW

**Comment by ericmjl** [comment]
Thank you, @kmaehashi!

File: tests/cupy_tests/logic_tests/test_type_test.py [hunk 1]
```diff
@@ -15,9 +17,66 @@ class TestBinaryRepr(testing.NumpyAliasBasicTestBase):
...
+
+class TypeTestingObjTestBase(object):
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test(self, xp, dtype):
+        return getattr(xp, self.func)(xp.ones(5, dtype=dtype))
+
+
+class TestIsComplex(TypeTestingTestBase, unittest.TestCase):
```
Comment (okuta) [inline tests/cupy_tests/logic_tests/test_type_test.py]: How about using parameterized test?

Comment (kmaehashi) [reply tests/cupy_tests/logic_tests/test_type_test.py]: Thanks! Fixed as suggested.

**Comment by okuta** [comment]
LGTM!

**Comment by okuta** [comment]
LGTM!

**Pull request merged**